patterns/1-initial/modular-code.md

```diff
@@ -8,54 +8,71 @@ The lack of modularization in the software architecture prevents reuseability, a
-    - Modular code makes Agile implementation easier.
-    - Modular efforts increase autonomy and accountability for Engineers, which can raise satisfaction and quality.
-* Select certain "success projects," teams that will develop reusable code and demonstrate the long term success. This can help motivate others (they see what is possible and what is in it for them). Transparency is critical.
-* Offer education. Modular code is well-understood; there is a lot of literature in favor of this.
-* Acknowledge and drive acceptance for the cost of modularization. Build time into the release schedule for this.
-* Companies moving to use more open source code will appreciate modularity more over time.
-* Mitigate risk and fear of quality degradation from accepting InnerSource contributions.
-* Establish a checklist of elements to be checked off to classify a component as reusable.
-    - There could be requirements on tests, tools and documentation before considering a component as reusable
-    - Introduce refactoring as organic effort and strive for it to become part of the culture. Refactoring leads to more modular code and reduction of technical debt.
-* Establish standards on testing methodology, labeling of repos.
+Modular code is far more reusable. Decoupling functionality into smaller narrow-focused modules eases
+and therefore promotes external contributions and onboarding of newcomers.
+
+* Offer education.
+  * On why: Lack of modularity is technical debt. It increases costs but hides them by shifting these into the future.
+  * On how: Modular code is well-understood; there is a lot of literature in favor of this.
+  * Select certain "success projects", teams that will develop reusable code and demonstrate the long term success. This can help motivate others (they see what is possible and what is in it for them). Transparency is critical.
+* Introduce refactoring as an organic effort and strive for it to become part of the culture.
+  * Acknowledge and drive acceptance for the investment in modularization. Build time into the release schedule for this.
+* Search for modularizable functionalities:
+  * Identify common functionalities and code blocks with potential for profitable modularity.
+  * Establish a checklist of elements to be checked off to classify a component as reusable.
+    There could be requirements on tests, tools and documentation before considering a component as reusable.
+    Establish standards on testing methodology, labeling of repos.
+* Eat the elephant in small bites. Break modularization down into baby steps, like
+  1. find internal would-be customers. Better ask them over guessing on your own.
+  1. define decoupling scope and preferred decoupling methods,
+  1. develop automated testing protection
+  1. document usage of modules.
+  1. introduce semantic versioning
+  1. use a dependency management software such as pip, nuget, npm
```

Comment:
```suggestion
  1. Find internal would-be customers. Better ask them over guessing on your own.
  1. Define decoupling scope and preferred decoupling methods.
  1. Develop automated testing protection.
  1. Document usage of modules.
  1. Introduce semantic versioning.
  1. Use a dependency management software such as pip, nuget, npm.
```

I think we should start all points with capital letters, to be consistent with the rest of the pattern.
Also added punctuation after each bullet point.

One question:

This is the only part of the pattern where we use a numbered list.
Is this on purpose? i.e. is there an order to this?

Reply:
@fioddor This is the only part of the pattern where we use a numbered list.
Is this on purpose? i.e. are these points meant to be used in this order?

Reply:
Arguably yes. I thought of a sequence, but it isn't the only possible one, so I don't have a strong opinion.

Reply:
Got it. Let's keep it as is then, and see if it sticks :)